javascript/ql/src/Security/CWE-020/UselessRegExpCharacterEscape.ql

```diff
@@ -0,0 +1,144 @@
+/**
+ * @name Useless regular expression character escape
+ * @description Prepending a backslash to a non-special character in a string
+ *              does not have any effect, and may make regular expressions
+ *              behave unexpectedly.
+ * @kind problem
+ * @problem.severity error
+ * @precision high
+ * @id js/useless-regexp-character-escape
+ * @tags correctness
+ *       security
+ *       external/cwe/cwe-20
+ */
+
+import javascript
+import semmle.javascript.CharacterEscapes::CharacterEscapes
+
+newtype TRegExpPatternMistake =
+  TIdentityEscapeInStringMistake(
+    RegExpPatternSource src, string char, string mistake, ASTNode rawStringNode, int index
+  ) {
+    char = getALikelyRegExpPatternMistake(src, mistake, rawStringNode, index)
+  } or
+  TBackspaceInStringMistake(RegExpPatternSource src, ASTNode rawStringNode, int index) {
+    exists(string raw |
+      rawStringNode = src.asExpr() and
+      raw = rawStringNode.(StringLiteral).getRawValue()
+      or
+      rawStringNode = src.asExpr().(TemplateLiteral).getAnElement() and
+      raw = rawStringNode.(TemplateElement).getRawValue()
+    |
+      "b" = getAnEscapedCharacter(raw, index) and
+      not (
+        // except if the string is exactly \b
```

Comment:
Could this logic be simplified by additionally keeping the (cooked) value of the string literal around?